awscrt/eventstream.py

```diff
@@ -195,3 +447,27 @@ def is_open(self):
             finished shutting down.
         """
         return _awscrt.event_stream_rpc_client_connection_is_open(self._binding)
+
+    def send_protocol_message(
+            self,
+            *,
+            message_type: EventStreamRpcMessageType,
+            headers: Optional[Sequence[EventStreamHeader]] = [],
+            payload: Optional[ByteString] = b'',
+            flags: int = EventStreamRpcMessageFlag.NONE) -> Future:
```

Comment:
For consistency with Java style, the order for sending or receiving messages is always headers, payload, messageType, messageFlags.

Reply:
tldr; doesn't matter

So python has the best argument parsing options in the world:
Everything after the `(self, *, ...)` is a "named keyword" argument, which is a new feature in python 3. It means the caller cannot pass these arguments by position alone, they MUST name each argument:
```
NOPE: connection.send_protocol_message(my_type, my_headers, my_payload, my_flags)
YEP:    connection.send_protocol_message(
                payload=my_payload,
                headers=my_headers,
                message_type=my_type,
                flags=my_flags)
```
The advantage of doing it this way is: you can make ALL kinds of API changes in a backwards compatible way. You can make an argument optional (which used to be required) by simply adding a default value. If the arguments were positional, that kind of change would be much more difficult, since python doesn't have function overloading.

Anyway, what I'm saying is: the order here doesn't matter. I made message_type the 1st argument because I made every other argument optional. But users can pass the arguments in any order they want, because they're naming the arguments, so the order doesn't matter.